### PR TITLE
Try to get_scalar_constant_value for Assert nodes

### PR DESCRIPTION
**Pull request by gvtulder**
If the conditions of an `Assert` are constant and True, `get_scalar_constant_value` could return the value if it it also constant.

### CODE REVIEW

File: theano/tensor/basic.py [hunk 1]
```diff
@@ -488,6 +488,14 @@ def get_scalar_constant_value(orig_v, elemwise=True,
...
                 v = v.owner.inputs[0]
                 continue
+            elif isinstance(v.owner.op, theano.tensor.opt.Assert):
+                # check if all conditions are constant and true
+                cond = [get_scalar_constant_value(c, max_recur=max_recur)
+                        for c in v.owner.inputs[1:]]
+                if builtins.all([0 == c.ndim and c != 0 for c in cond]):
+                    v = v.owner.inputs[0]
+                # not all conditions are satisfied
+                continue
```
Comment (nouiz) [inline theano/tensor/basic.py]: The indentation of this line is not good. Add 4 spaces before continue.

Otherwise, this PR look good. Why did you need this case?

Comment (gvtulder) [inline theano/tensor/basic.py]: Changed now.

**Comment by gvtulder** [comment]
I ran into this when I added an Assert in #5267. There I tried to test a shape tuple by wrapping it in an Assert, before passing it to `gpu_alloc_empty`. It turns out that `gpu_alloc_empty` uses `get_scalar_constant_value` to determine the broadcast pattern of the new array. Wrapping a constant in an Assert broke that bit.
I've now solved my original problem in a better way by testing constant values at compile time, but this would also have solved it. It might still be useful in other cases.
